### PR TITLE
Feat: automate package publishing using semantic-release

### PR DESCRIPTION
**Pull request by BalbinaK**
### Description
1. Adds a github action that triggers a release with semantic-release whenever a commit is pushed to `alpha` or `main` branch. The result of the Release job is the following:
- the package will be published to NPM with whatever files are built to the `dist` folder, as well as `package.json`, `LICENSE` (once we get one), and `README.md`
- a github release will be created with a corresponding tag version and will include a `release.tgz` tarball
   a) a merge to `alpha` branch will create a release of `X.X.X-alpha.X` version (e.g. 1.1.0-alpha.1)
   b) a merge of `alpha` to `main` will create a release of `X.X.X` version (e.g. 1.1.0)
- semantic-release-bot will push a commit with a respective updates to `CHANGELOG.md` and `package.json`

2. We are using conventional commits. The version published will depend on the [commit message structure](https://github.com/semantic-release/semantic-release#commit-message-format).

3. The `NPM_TOKEN` was added to `@warp-ds` organisation secrets in Github but will eventually be replaced with a token created by a warp-bot-user.

### Notes
To ensure that automatic release works properly, we might want to set up some rules/processes for development. E.g.:
- allow squashing to `alpha` BUT ensure PR titles follow the [commit message structure](https://github.com/semantic-release/semantic-release#commit-message-format) 
- only possible to merge to `main` from `alpha` with a PR
- squashing or pushing directly to `main` is blocked
- after merging `alpha` to `main` an auto-merging github action is triggered - `alpha` should always be up to date with latest `main`

Co-authored-by: Anna Rybkina <anna.rybkina@anicode.se>
Co-authored-by: Magnus Hübner <magnus.hubner@aptitud.se>

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: current
+      - uses: pnpm/action-setup@v2
```
Comment (pearofducks) [inline .github/workflows/release.yml]: I suggest either all steps have `name` or only the ones that are 'user controlled'. A bit weird to me the two steps above that will pretty much never fail (and if they do there's nothing we can do about it) are named, but this step that could doesn't have a name. 🙈

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          version: 7
+          run_install: true
+      - name: Install dependencies
```
Comment (pearofducks) [inline .github/workflows/release.yml]: This step is not needed, pnpm automatically does that above with the `run_install` flag

File: LICENSE [hunk 3]
```diff
@@ -1,21 +0,0 @@
-MIT License
```
Comment (pearofducks) [inline LICENSE:1]: This should be modified to the Apache license, not removed

File: .github/workflows/release.yml (hunk 2 above)
Comment (BalbinaK) [reply .github/workflows/release.yml]: Good point! We fixed it in [d3091ee](https://github.com/warp-ds/core/pull/3/commits/d3091ee6951ef423d4c8504f5b59a2c65dc6f9c9)

File: .github/workflows/release.yml (hunk 1 above)
Comment (BalbinaK) [reply .github/workflows/release.yml]: Thanks! We forgot about the name and although those are not user controller, we felt a name helps understand what is happening step by step in the workflow. [d3091ee](https://github.com/warp-ds/core/pull/3/commits/d3091ee6951ef423d4c8504f5b59a2c65dc6f9c9)

File: LICENSE (hunk 3 above)
Comment (BalbinaK) [reply LICENSE:1]: Fixed in [7799a7a](https://github.com/warp-ds/core/pull/3/commits/7799a7abe0a4fee433b5ebf201ce6a69887445d3) and added a "license" entry to `package.json` in [ba4a93d](https://github.com/warp-ds/core/pull/3/commits/ba4a93ded42e80e50707db960325a6c3c9ff7d37). 

We also noticed that the [license in warp drive](https://github.com/warp-ds/drive/blob/master/package.json#L31) was `"MIT"`, but maybe it should also be `"Apache-2.0"`. Should we update it?

Comment (pearofducks) [reply LICENSE:1]: Yes update please!

**Review by pearofducks — APPROVED** [review]

**Comment by pearofducks** [comment]
Once we are done with the initial development - we'll change to merging straight into `main` from a PR, correct?

(Ofc one _can_ make an alpha release if there's a more major feature being developed, but it wouldn't be the default way of doing things?)

**Comment by magnuh** [comment]
> Once we are done with the initial development - we'll change to merging straight into `main` from a PR, correct?
> 
> (Ofc one _can_ make an alpha release if there's a more major feature being developed, but it wouldn't be the default way of doing things?)

Yes, only alpha until we are happy about the 1.0.0 release... just to not mess up the package version history in npm. 😄 

...and then I guess we should discuss if we need the `next` thing (from the react repo) or not.

**Comment by pearofducks** [comment]
Makes sense, thanks!

**Review by trygve-lie — APPROVED** [review]
This looks great! Can't see anything off here. Great work everyone :100:

**Comment by BalbinaK** [comment]
[Automate releases of warp packages to NPM and EIK](https://trello.com/c/v1Ptqg8G/128-automate-releases-of-warp-packages-to-npm-and-eik)

**Pull request merged**